src/main.cpp

```diff
@@ -74,23 +74,6 @@ static string get_generator_string() {
       .str();
 }
 
-/**
- * convert an environment variable name to an option name
- */
-static string environment_option_name(string name){
-  string option;
-
-  if (name.substr(0, 7) == "CGIMAP_") {
-    std::transform(name.begin() + 7, name.end(),
-                   std::back_inserter(option),
-                   [](unsigned char c) {
-                     return c == '_' ? '-' : std::tolower(c);
-                   });
-  }
-
-  return option;
-}
-
```

Comment:
Replaced with lambda expression below that also validates against known option names.